### ISSUE TITLE
Parameterized message type 

### ISSUE THREAD

**Issue by nossleinad** [MurrellGroup/MolecularEvolution.jl#27]
We should consider a parameterized message type for `FelNode`. The reason for this is to avoid conversions from `Vector{PType} where PType <: Partition` (and PType != Partition) to `Vector{Partition}` and [Avoid containers with abstract type parameters](https://docs.julialang.org/en/v1/manual/performance-tips/#man-performance-abstract-container).

**Comment by murrellb** [comment]
Good idea. We need to make sure that the implementation of this doesn't hurt generality though, or make it too annoying to manipulate the message (like adding an extra partition to the end of an existing message). Needs some thought.

**Comment by nossleinad** [comment]
There is a similar issue for BWMModel
